NMS-14084: Add the ability to define an enumeration to convert collected strings into numeric values (#4723)
OpenNMS/opennms: docs/modules/operation/pages/performance-data-collection/snmp-property-extenders/value-mapping.adoc

= Value Mapping Property Extender
The Value Mapping property extender provides a mechanism that works as a lookup table for converting collected values to a different data type.

> This allows you to do things such as map an enumerated set of string values to a corresponding set of integer values.

Reviewer: ```suggestion
This allows you to do things such as map an enumerated set of string values to a corresponding set of integer values or the other way around.
```

= Value Mapping Property extender
The Value Mapping property extender provides a mechanism that works like a lookup table for values of a local MIB table column.

== When to use the Value Mapping Property extender
Use the Value Mapping property extender to map an enumerated set of values (e.g. strings) to a corresponding set of values.

> This allows to persist changes of string enumeration values as integers over time as time-series.

Reviewer 2: ```suggestion
This allows OpenNMS to persist changes of string enumeration values as integers over time as time-series.
```
Object was missing. Not sure if OpenNMS is the right word here.

Author: Yes. It is.

> The Value Mapping property extender provides a mechanism that works like a lookup table for values of a local MIB table column.
> 
> == When to use the Value Mapping property extender
> Use the Value Mapping property extender to map an enumerated set of values (for example, strings) to a corresponding set of values.
> This allows OpenNMS to persist changes of string enumeration values as integers over time as time-series.

Reviewer 3: ```suggestion
The Value Mapping property extender provides a mechanism that works as a lookup table for converting collected values to a different data type.
This allows you to do things such as map an enumerated set of string values to a corresponding set of integer values.
The resulting values are then persisted as time-series data to track state changes over time.
```

> == Configure the Value Mapping Property Extender

Reviewer 2: ```suggestion
== Configure the Value Mapping property extender
```

The Value Mapping property extender expects zero or more parameters.
There are currently three parameters with a fixed name:

.Value Mapping Property Extender Parameters
[options="header", cols="1,3,2"]
|===
| Name
| Description
| Default value

3+| *Required*

| source-attribute
| The alias name of the property used for the value lookup.
| n/a

| target-type

> | The target data-type, e.g. gauge.

Reviewer 2: ```suggestion
| The target data-type; for example, gauge.
```
This is where I think the "for example" rule is annoying. :-)
